packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, CheckboxControl, Modal } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useCallback } from '@wordpress/element';
+
+const EntitiesSavedStatesCheckbox = ( {
+	id,
+	name,
+	changes: { rawRecord },
+	checked,
+	setCheckedById,
+} ) => (
+	<CheckboxControl
+		label={ `${ startCase( name ) }: "${ rawRecord.title || rawRecord.name }"` }
+		checked={ checked }
+		onChange={ useCallback( ( nextChecked ) => setCheckedById( id, nextChecked ), [ id ] ) }
+	/>
+);
+
+export default function EntitiesSavedStates( { isOpen, onRequestClose } ) {
+	const entityRecordChangesByRecord = useSelect( ( select ) =>
+		select( 'core' ).getEntityRecordChangesByRecord()
+	);
+	const { saveEditedEntityRecord } = useDispatch( 'core' );
+
+	const [ checkedById, _setCheckedById ] = useState( {} );
```

Comment:
Not really a blocker, and maybe discussed previously, but wouldn't it be more expected that saving is an opt-out, not an opt-in?

Reply:
I think we should only check the current post by default, but we could leave that for another PR. There was also discussion around how drafts should/should not be checked.